src/pydvl/influence/types.py

```diff
@@ -1,48 +1,53 @@
 from abc import ABC
-from typing import Callable, Iterable, Optional, Tuple
+from typing import TYPE_CHECKING, Iterable, Optional, Tuple, Union
 
 from numpy import ndarray
 
-__all__ = [
-    "TwiceDifferentiable",
-    "MatrixVectorProduct",
-    "MatrixVectorProductInversionAlgorithm",
-]
+try:
+    import torch
+    import torch.nn as nn
+
+    _TORCH_INSTALLED = True
+except ImportError:
+    _TORCH_INSTALLED = False
+
+if TYPE_CHECKING:
+    from numpy.typing import NDArray
```

Comment:
We decided in a previous discussion to import this directly instead of only during type checking because it doesn't add anything to do it this way.